src/abbfreeathome/devices/cover_actuator.py

```diff
@@ -0,0 +1,261 @@
+        return self.state == 2
+
+    def is_cover_closing(self):
+        """Helper-Function returns true if the cover is closing."""
+        return self.state == 3
+
+    async def open(self):
+        """Open the cover."""
+        await self._set_moving_datapoint("0")
+
+    async def close(self):
+        """Close the cover."""
+        await self._set_moving_datapoint("1")
+
+    async def stop(self):
+        """Stop the movement of the cover."""
+        if self.state in [2, 3]:
+            await self._set_stop_datapoint()
+
+    async def force_position(self, value: int):
+        """
+        Force the position of the cover.
+
+        0 means none
+        2 means open
+        3 means close
+        """
+        if value in [0, 2, 3]:
+            await self._set_force_datapoint(str(value))
+
+    async def set_position(self, value: int):
+        """
+        Set the position of the cover.
+
+        The position has to be between 0 and 100
+        Fully open = 0
+        Fully closed = 100
+        Just as an information: This is exaclty the other way round as done in HA,
+        so in HA we have to remember to convert the value with something like:
+        abs(value-100)
+        before sending it to this function
+        """
+        value = max(0, value)
+        value = min(value, 100)
+
+        await self._set_position_datapoint(str(value))
+        self._position = value
+
+    def _refresh_state_from_output(self, output: dict[str, Any]) -> bool:
+        """
+        Refresh the state of the device from a given output.
+
+        This will return whether the state was refreshed as a boolean value.
+        """
+        if output.get("pairingID") == Pairing.AL_INFO_MOVE_UP_DOWN.value:
+            """
+            0 means the blind is fully opened
+            1 means the blind is partly opened
+            2 means the blind is currently opening
+            3 means the blind is currently closing
+            """
+            self._state = int(output.get("value"))
+            return True
+        if output.get("pairingID") == Pairing.AL_INFO_FORCE.value:
+            """
+            0 means none
+            2 means open
+            3 means close
+            """
+            self._forced_position = int(output.get("value"))
+            return True
+        if (
+            output.get("pairingID")
+            == Pairing.AL_CURRENT_ABSOLUTE_POSITION_BLINDS_PERCENTAGE.value
+        ):
+            self._position = int(output.get("value"))
+            return True
+        if (
+            output.get("pairingID")
+            == Pairing.AL_CURRENT_ABSOLUTE_POSITION_SLATS_PERCENTAGE.value
+        ):
+            self._tilt_position = int(output.get("value"))
+            return True
+        if (
+            output.get("pairingID")
+            == Pairing.AL_CURRENT_ABSOLUTE_POSITION_SLATS_PERCENTAGE.value
+        ):
+            self._tilt_position = int(output.get("value"))
+            return True
```

Comment:
This is duplicated, causing your tests to never pass as it's never getting to the second if.

```suggestion
```

Reply:
Oh, stupid me.

Sometimes I should stop working too late 😢